### ISSUE TITLE
Fetching Audit Logs leads to Uncaught Promise Error if rejected

### ISSUE THREAD

**Issue by CharlotteDunois** [hydrabolt/discord.js#1471]
**Please describe the problem you are having in as much detail as possible:**
Using `<Guild>.fetchAuditLogs` leads to an uncaught promise error, if the request is rejected by the Request Handler (like due to missing permissions).

Using following piece:
```js
message.guild.fetchAuditLogs({ user: '200317799350927360' }).catch((err) => console.log('hydar', err))
```

Logs:
```
1|CharuruD | 2017/05/07, 00:57:05.733@ Uncaught Promise Error:
1|CharuruD | { DiscordAPIError: Missing Permissions
1|CharuruD |     at item.request.gen.end (/var/opt/Discord_Bot/node_modules/discord.js/src/client/rest/RequestHandlers/Sequential.js:68:65)
1|CharuruD |     at then (/var/opt/Discord_Bot/node_modules/snekfetch/src/index.js:181:21)
1|CharuruD |     at process._tickCallback (internal/process/next_tick.js:109:7)
1|CharuruD |   name: 'DiscordAPIError',
1|CharuruD |   message: 'Missing Permissions',
1|CharuruD |   code: 50013 }
```

**Further details:**

- discord.js version: v11.1 master 07623d49e251b462a2e2f22cae31d3c8dfcb090d
- node.js version: v7.9.0
- Operating system: Debian jessie
- Priority this issue should have – please be realistic and elaborate if possible: Normal

**Comment by dragonfire535** [comment]
You probably do not have permission to View Audit Logs.

**Comment by appellation** [comment]
The issue, as reported, is that the lib is not correctly passing down the promise rejection.

**Comment by dragonfire535** [comment]
Oh. Welp.

**Comment by iCrawl** [comment]
Fixed in f95f18b5 on master and bb0ee597 on v11.2 respectively.
